http/src/main/java/io/micronaut/http/HttpRequest.java

```diff
@@ -179,7 +181,28 @@ default Optional<Locale> getLocale() {
      */
     @SuppressWarnings("deprecation")
     default Optional<Certificate> getCertificate() {
-        return this.getAttribute(HttpAttributes.X509_CERTIFICATE, Certificate.class);
+        Optional<Certificate> attribute = this.getAttribute(HttpAttributes.X509_CERTIFICATE, Certificate.class);
```

Comment:
When is this ever present?  Noticed that this being present isn't covered by the tests, and just wondered as it's the only usage of this deprecated constant 🤔

Reply:
unfortunately HttpAttributes.X509_CERTIFICATE is public api so a user couldve set the attribute themselves